### PR TITLE
Don't impose WebSockets limits on Extensions

### PR DESCRIPTION
**Pull request by goodov**
Resolves https://github.com/brave/brave-browser/issues/21661

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: chromium_src/third_party/blink/renderer/modules/websockets/websocket_channel_impl.h [hunk 1]
```diff
@@ -30,7 +30,7 @@ using WebSocketChannelImpl_BraveImpl = WebSocketChannelImpl;
 
 namespace blink {
 
-class MODULES_EXPORT WebSocketChannelImpl
+class MODULES_EXPORT WebSocketChannelImpl final
```
Comment (goodov) [inline chromium_src/third_party/blink/renderer/modules/websockets/websocket_channel_impl.h:33]: here are follow up fixes from the original PR https://github.com/brave/brave-core/pull/11609

File: chromium_src/third_party/blink/renderer/modules/websockets/websocket_channel_impl.cc [hunk 2]
```diff
@@ -39,7 +40,10 @@ bool WebSocketChannelImpl::ShouldDisallowConnection(const KURL& url) {
   if (base::FeatureList::IsEnabled(blink::features::kRestrictWebSocketsPool)) {
     if (blink::WebContentSettingsClient* settings =
             brave::GetContentSettingsClientFor(execution_context_)) {
-      if (settings->GetBraveFarblingLevel() != BraveFarblingLevel::OFF) {
+      const bool is_extension = CommonSchemeRegistry::IsExtensionScheme(
+          execution_context_->GetSecurityOrigin()->Protocol().Ascii());
+      if (!is_extension &&
```
Comment (goodov) [inline chromium_src/third_party/blink/renderer/modules/websockets/websocket_channel_impl.cc:45]: this is the main change in this PR.

File: browser/brave_shields/websockets_pool_limit_browsertest.cc [hunk 3]
```diff
@@ -302,6 +309,31 @@ IN_PROC_BROWSER_TEST_F(WebSocketsPoolLimitBrowserTest,
...
+#if BUILDFLAG(ENABLE_EXTENSIONS)
+IN_PROC_BROWSER_TEST_F(WebSocketsPoolLimitBrowserTest,
+                       PoolIsNotLimitedForExtensions) {
+  extensions::TestExtensionDir test_extension_dir;
+  test_extension_dir.WriteManifest(R"({
+    "name": "Test",
+    "manifest_version": 2,
+    "version": "0.1",
+    "permissions": ["webRequest", "webRequestBlocking", "*://a.com/*"],
+    "content_security_policy": "script-src 'self' 'unsafe-eval'; object-src 'self'"
```
Comment (goodov) [inline browser/brave_shields/websockets_pool_limit_browsertest.cc:321]: `unsafe-eval` required for `EvalJs` to work.

**Review by mkarolin — APPROVED** [review]
LGTM

**Pull request merged**